core/src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,10 +1,27 @@
 package pureconfig
 
+/**
+ * A mapping between case class fields and their respective keys in the config.
+ */
 trait ConfigFieldMapping extends (String => String) {
+
   def apply(fieldName: String): String
+
+  /**
+   * Returns a `ConfigFieldMapping` that uses this mapping with some exceptions.
+   *
+   * @param exceptions the exceptions for this mapping as pairs (field, configKey)
```

Comment:
I prefer `overrides`. While I agree on the meaning of exception in this case, I also think the name can be confusing. I like override because it can be seen as "override the mapping behavior for the given fields"